cmd/promtool/backfill.go

```diff
@@ -93,7 +93,10 @@ func createBlocks(input []byte, mint, maxt int64, maxSamplesInAppender int, outp
 		nextSampleTs = math.MaxInt64
 
 		err := func() error {
-			w, err := tsdb.NewBlockWriter(log.NewNopLogger(), outputDir, blockDuration)
+			// 2*blockDuration block size keep the appendable window to blockDuration
+			// irrespective of which timestamp is added within the blockDuration range.
+			// The final block will be at max blockDuration since it depends on the samples.
```

Comment:
That doesn't quite parse. Perhaps insert a "to" between "size" and "keep"?

Since this is a bit hard to understand in general, it might also be better to explain more. Suggestion:

"To prevent races with compaction, a block writer only allows appending samples that are at most half a block size older than the most recent sample appended so far. However, in the way we use the block writer here, compaction doesn't happen, while we also need to append samples throughout the whole block range. To allow that, we  pretend that the block is twice as large here, but only really add sample in the original interval later."